### PR TITLE
[Functor Task(1)] DKishmar

### PR DESCRIPTION
**Pull request by DKishmar**

### CODE REVIEW

File: functors task (1)/student/include/impl.h [hunk 1]
```diff
@@ -8,6 +8,16 @@
  */
 struct BeerOrganizer
 {
+    int x = 1;
+
+    BeerBrand operator()()
+    {
+       int currentBeer = x;
+       if(x < 6) x++;
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:16]: **avoid magic numbers!**

File: functors task (1)/student/include/impl.h [hunk 2]
```diff
@@ -8,6 +8,16 @@
  */
 struct BeerOrganizer
 {
+    int x = 1;
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:11]: use descriptive names!

File: functors task (1)/student/include/impl.h [hunk 3]
```diff
@@ -8,6 +8,16 @@
...
+
+    BeerBrand operator()()
+    {
+       int currentBeer = x;
+       if(x < 6) x++;
+       else x = 1;
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:17]: don't write two expressions in one line
also, follow this rule:
https://wiki.c2.com/?AlwaysUseBracesOnIfThen

File: functors task (1)/student/include/impl.h [hunk 4]
```diff
@@ -16,16 +26,18 @@ struct BeerOrganizer
  *
  * @note Only Corona and HoeGaarden are expensive
  */
-bool isExpensiveBeer(/**???*/)
+bool isExpensiveBeer(BeerBrand & brand)
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:29]: scalar types are usually passed by copy
also, if you don't modify params, use const

File: functors task (1)/student/include/impl.h [hunk 5]
```diff
@@ -40,9 +52,27 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
+    static Cocktail mix(AlcoholDrink alcohol, NonAlcoholDrink noAlcohol)
     {
+        Cocktail currentCoctail = Cocktail::Oops;
+        if(alcohol == AlcoholDrink::Gin)
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:58]: _Nice, but consider switch_

**Review by smay1613 — CHANGES_REQUESTED** [review]

File: functors task (1)/student/include/impl.h [hunk 6]
```diff
@@ -8,6 +8,24 @@
  */
 struct BeerOrganizer
 {
+    int firstBeer = static_cast<int>(BeerBrand::HoeGaarden);
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:11]: _prefer size_t_

File: functors task (1)/student/include/impl.h (hunk 1 above)
Comment (DKishmar) [reply functors task (1)/student/include/impl.h]: fixed

File: functors task (1)/student/include/impl.h (hunk 2 above)
Comment (DKishmar) [reply functors task (1)/student/include/impl.h]: fixed

File: functors task (1)/student/include/impl.h (hunk 4 above)
Comment (DKishmar) [reply functors task (1)/student/include/impl.h]: fixed

File: functors task (1)/student/include/impl.h (hunk 3 above)
Comment (DKishmar) [reply functors task (1)/student/include/impl.h]: fixed

File: functors task (1)/student/include/impl.h (hunk 6 above)
Comment (DKishmar) [reply functors task (1)/student/include/impl.h]: fixed

File: functors task (1)/student/include/impl.h [hunk 7]
```diff
@@ -8,6 +8,24 @@
  */
 struct BeerOrganizer
 {
+    size_t firstBeer = static_cast<int>(BeerBrand::HoeGaarden);
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:11]: prefer brace initialization
you forgot about static_cast<int> - you can avoid such kind of errors with "{}" init

**Pull request closed without merging**